multiversx_sdk_core/transaction_factories/smart_contract_factory.py

```diff
@@ -107,44 +92,31 @@ def upgrade(self,
 
         parts += args_to_strings(arguments)
 
-        transaction = self.create_transaction(
+        transaction = self._create_transaction_intent(
             sender=sender,
             receiver=contract,
             data_parts=parts,
-            gas_limit=gas_limit,
-            nonce=nonce,
-            guardian=guardian
+            gas_limit=gas_limit
         )
 
         return transaction
 
-    def create_transaction(
+    def _create_transaction_intent(
             self,
             sender: IAddress,
             receiver: IAddress,
             data_parts: List[str],
-            gas_limit: IGasLimit,
-            nonce: Optional[INonce],
-            value: Optional[ITransactionValue] = None,
-            guardian: Optional[IAddress] = None,
-            gas_price: Optional[IGasPrice] = None) -> Transaction:
+            gas_limit: int) -> TransactionIntent:
         data = self._build_transaction_payload(data_parts)
-        version = TRANSACTION_VERSION_DEFAULT
-        options = TRANSACTION_OPTIONS_DEFAULT
 
-        return Transaction(
-            chain_id=self.config.chain_id,
-            sender=sender,
-            receiver=receiver,
-            gas_limit=gas_limit,
-            gas_price=gas_price,
-            value=value,
-            nonce=nonce,
-            data=data,
-            version=version,
-            options=options,
-            guardian=guardian
-        )
+        transaction_intent = TransactionIntent()
+
+        transaction_intent.sender = sender.bech32()
+        transaction_intent.receiver = receiver.bech32()
+        transaction_intent.gas_limit = gas_limit
+        transaction_intent.data = str(data).encode()
+
+        return transaction_intent
 
     def _build_transaction_payload(self, parts: List[str]) -> TransactionPayload:
         data = ARGS_SEPARATOR.join(parts)
```

Comment:
This code will be replicated in every factory, maybe extracted in a TransactionIntendBuilder or something similar